src/cider/nrepl/middleware/info.clj

```diff
@@ -280,23 +280,34 @@
                             (sort-by count))
     :else (:arglists info)))
 
-(defn format-eldoc [raw-eldoc]
-  (map #(mapv str %) raw-eldoc))
+(defn format-arglists [raw-arglists]
+  (map #(mapv str %) raw-arglists))
+
+(defn extract-ns
+  [info]
+  (cond
+    (:ns info) (:ns info)
+    (:candidates info) (->> (:candidates info)
+                            keys)
+    :else (:class info)))
```

Comment:
This method treats classes as arrays. We could also return a `:class` key. Would that be better ?